Backend/Meta-TV2-api/Meta-TV2-BusinessLayer/BusinessRules.cs

```diff
@@ -161,4 +162,66 @@ public class BusinessRules : IBusinessRules
             return false;
         }
     }
+
+    public async Task<string> GetPosts() {
+        try {
+            var posts = await DataAccess.GetPosts();
+            if (!posts.HasValue)
+                return null;
+            return JsonSerializer.Serialize(posts.Value);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    public async Task<string> GetPosts(int id) {
+        try {
+            var posts = await DataAccess.GetPosts(id);
+            if (!posts.HasValue)
+                return null;
+            return JsonSerializer.Serialize(posts.Value);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    public async Task<(string, string)> GetPostFileType(int id) {
+        try {
+            var post = await DataAccess.GetPostByPostId(id);
+            if (!post.HasValue)
+                return (null, null);
+            if(post.Value.pathType == "Url")
+                return ("Url", post.Value.filePath);
+            return (post.Value.pathType, post.Value.filePath);
+        } catch (Exception e){
+            return (null, null);
+        }
+    }
+
+    public async Task<bool> AddPost(string post, ICustomFormFile file)
+    {
+        try {
+            var deserializedPost = JsonSerializer.Deserialize<Posts>(post);
+            if (file == null | deserializedPost.pathType == "")
+                return false;
+            if(deserializedPost.pathType == "Url" & deserializedPost.filePath != "") { // Handle URL case
+                DataAccess.AddPostWithUrl(deserializedPost);
+                return true;
+            }
+            deserializedPost.filePath = "." + file.ContentType.Split("/")[1];
+            int id = await DataAccess.AddPostWithFile(deserializedPost);
+            if (id == -1)
+                return false;
+
+            var path = Path.Combine(Directory.GetCurrentDirectory(), "wwwroot", deserializedPost.pathType, id.ToString() + "." + file.ContentType.Split("/")[1]);

```

Comment:
Den här är ju lite farlig; om jag förstått det rätt så kan man sätta `pathType` till vad som helst och om man sätter det till att börja med ett `/` så ignoreras de tidigare två argumenten och då får man spara filer lite var som helst

Reply:
Hej, syftar på du kanske på pathType? Eftersom det är det enda som skulle kunna påverka var filen hamnar utifrån frontend, resten av pathen avgörs av backend via ett ID från databasen ihopslaget med pathType och ContentType, därav det sist nämnda är tex .jpg osv.

Reply:
Should be resolved in latest push, added hard check for pathType